File: lua/types/responses.lua
---@class SpellingProblem
---@field lineStart number
---@field lineOfset number
---@field word string

---@class SpellCheckResponse
---@field kind "lint" # Literal string type
---@field problems SpellingProblem[] # Array of SpellingProblem

---@class LintError
---@field kind "error" # Literal string type
---@field message string

---@alias SpellResponse SpellCheckResponse|LintError

File: jslib/src/types/responses.ts

interface SpellingProblem{
    lineStart: number,
    lineOfset: number,
    word: string
    
}

interface SpellCheckResponse{
    kind: "lint"
    problems: Array<SpellingProblem>
}

interface SuggestionResponse {
    kind: "suggestion",
    suggestion: Array<string>
}

interface ErrorResponse{
    kind: "error",
    message: string
}

type SpellResponse = SpellCheckResponse | ErrorResponse | SuggestionResponse;

export {SpellResponse, SpellCheckResponse, SuggestionResponse}
